### PR TITLE
feat: add `HookParameters` type to integrations

### PR DESCRIPTION
**Pull request by ematipico**
#### Description (required)

This PR improves the reference of the integrations, by adding the types to each hook.

Astro exposes a utility type called `HookParameters<'<HOOK_NAME>'>` that allows for easy type inference.

I took the opportunity to change the code block to `ts` when we show types in the code snippets.

### CODE REVIEW

File: src/content/docs/en/reference/integrations-reference.mdx [hunk 1]
```diff
@@ -19,6 +19,8 @@ The official Astro integrations can act as reference for you as you go to build
 ## Quick API Reference
 
 ```ts
+import { HookParameters } from 'astro';
```
Comment (Princesseuh) [inline src/content/docs/en/reference/integrations-reference.mdx:22]:
```suggestion
import type { HookParameters } from 'astro';
```

File: src/content/docs/en/reference/integrations-reference.mdx [hunk 2]
```diff
@@ -81,6 +83,7 @@ The following hooks are built in to Astro:
 **Why:** To extend the project config. This includes updating the [Astro config](/en/reference/configuration-reference/), applying [Vite plugins](https://vitejs.dev/guide/api-plugin.html), adding component renderers, and injecting scripts onto the page.
 
 ```ts
+import {HookParameters} from "astro";
```
Comment (ascorbic) [inline src/content/docs/en/reference/integrations-reference.mdx:86]:
```suggestion
import type { HookParameters } from "astro"; 
```

**Comment by ascorbic** [comment]
Nice improvement. I only discovered this type this week!

**Review by delucis — COMMENTED** [review]
I don’t think the way `HookParameters` has been added here is valid TypeScript, is it? You can’t type the parameter twice — you‘re basically doing: `(options: TypeA: TypeB) => ...`

I think I might suggest a separate heading rather than shoe-horning the type into this code block. That also aligns better with usage I think to explain when you might want `HookParameters`.

**Comment by sarah11918** [comment]
Just saying that I am going to let @delucis be the one to give final approval here!

File: src/content/docs/en/reference/integrations-reference.mdx [hunk 3]
```diff
@@ -393,13 +396,13 @@ The **`stage`** denotes how this script (the `content`) should be inserted. Some
...
 
-```js
+```ts
 'astro:config:done'?: (options: {
   config: AstroConfig;
   setAdapter: (adapter: AstroAdapter) => void;
   injectTypes: (injectedType: { filename: string; content: string }) => URL;
   logger: AstroIntegrationLogger;
-}) => void | Promise<void>;
+}: HookParameters<'astro:config:done'>) => void | Promise<void>;
```
Comment (Fryuni) [inline src/content/docs/en/reference/integrations-reference.mdx:405]: It should remove the previous type, now it is `varName: oneType: anotherType`

Same with all of them

**Review by Fryuni — CHANGES_REQUESTED** [review]

**Review by delucis — COMMENTED** [review]
Thanks again for kicking this off @ematipico! Cleaned this up to add a dedicated section for `HookParameters` to help explain how to use it.

**Review by sarah11918 — APPROVED** [review]

**Pull request merged**